lib/async_hooks.js

```diff
@@ -461,6 +462,18 @@ function init(asyncId, type, triggerId, resource) {
   processing_hook = false;
 }
 
+const JSProviders = new Set();
+function registerTypeName(type) {
+  if (JSProviders.has(type))
```

Comment:
Maybe we should make sure `type` isn't already contained in `async_wrap.Providers` either?